syntax = "proto2";

import "protos/model.proto";
import "protos/hyperparams.proto";


message AdViSEGCNV4Model {
  extend Model {
    optional AdViSEGCNV4Model ext = 1423;
  }

  // If true, also add image as a ROI.
  optional bool add_image_as_a_roi = 1 [default = true];

  // Hyper-parameters of fully-connected layer.
  optional Hyperparams fc_hyperparams = 2;

  // Path to the GloVe file.
  optional string glove_path = 3;

  // Slogan vocab file.
  optional string slgn_vocab_list_path = 4;

  // Statement vocab file.
  optional string stmt_vocab_list_path = 5;

  // Word embedding dimensions.
  optional int32 embedding_dims = 6 [default = 200];

  // Inference dimensions.
  optional int32 inference_dims = 7 [default = 100];

  // Dropout keep probability of the multi-modal dot-product operation.
  optional float dropout_keep_prob = 8 [default = 0.5];

  // Margin of triplet loss.
  optional float triplet_margin = 9 [default = 0.2];

  // If true, use reduce_mean to aggregate the distance between anchor and positive example.
  optional bool triplet_ap_use_avg = 10 [default = false];

  // If true, connect roi to slogan nodes.
  optional bool connect_roi_to_slogan = 21 [default = false];
}
